### PR TITLE
feat: Support Casbin UpdatableAdapter interface

### PR DESCRIPTION
**Pull request by basakest**

### CODE REVIEW

**Comment by hsluoyz** [comment]
@techoner plz review

File: src/Adapter.php [hunk 1]
```diff
@@ -187,7 +187,33 @@ public function removeFilteredPolicy(string $sec, string $ptype, int $fieldIndex
...
 
+        $removedRules = $this->casbinRule->find()->where($where)->all();
         $this->casbinRule->deleteAll($where);
+
+        array_walk($removedRules, function (&$removedRule) {
+            unset($removedRule->id);
+            $removedRule = $removedRule->toArray();
+            $removedRule = array_filter($removedRule, function ($value) {
+                return !is_null($value) && $value !== '';
+            });
```
Comment (techoner) [inline src/Adapter.php:198]: @basakest For filtering rules, please refer to: https://github.com/php-casbin/dbal-adapter/blob/master/src/Adapter.php#L435

File: src/Adapter.php [hunk 2]
```diff
@@ -226,6 +250,98 @@ public function loadFilteredPolicy(Model $model, $filter): void
...
+    public function updateFilteredPolicies(string $sec, string $ptype, array $newRules, int $fieldIndex, ?string ...$fieldValues): array
+    {
+        $oldRules = [];
+        $transaction = $this->casbinRule->getDb()->beginTransaction();
+        try {
+            $oldRules = $this->_removeFilteredPolicy($sec, $ptype, $fieldIndex, ...$fieldValues);
+            $this->addPolicies($sec, $ptype, $newRules);
+            $transaction->commit();
+        } catch (\Exception $e) {
+            $transaction->rollBack();
```
Comment (techoner) [inline src/Adapter.php:319]: @basakest  The exception should not be ignored here, same as updatePolicies().

**Review by techoner — CHANGES_REQUESTED** [review]

**Review by techoner — APPROVED** [review]

**Pull request merged**